### PR TITLE
Quarkus bom manage smallrye-reactive-messaging-api

### LINKED ISSUE #9033: Quarkus-bom should manage smallrye reactive-messaging-api dependency
**Description**
Smallrye reactive-messaging-api is not inside quarkus-bom, so I have to manage the dependency on my build.
However when smallrye-reactive-messaging extension switched to Smallrye reactive-messaging-api 2.0.1 with #8565, some of my integration test did not pass because I was still using the old version.

**Implementation ideas**
quarkus-bom should manage the smallrye-reactive-messaging-api it uses.

Regards
Kevin

### PR DESCRIPTION
**Pull request by vietk**
Fixes #9033

### CODE REVIEW

File: bom/runtime/pom.xml [hunk 1]
```diff
@@ -2545,6 +2545,11 @@
                 <version>${quartz.version}</version>
             </dependency>
 
+            <dependency>
+                <groupId>io.smallrye.reactive</groupId>
+                <artifactId>smallrye-reactive-messaging-api</artifactId>
+                <version>${smallrye`-reactive-messaging.version}</version>
```
Comment (gsmet) [inline bom/runtime/pom.xml:2551]:
```suggestion
                <version>${smallrye-reactive-messaging.version}</version>
```

**Review by gsmet — CHANGES_REQUESTED** [review]

**Review by geoand — APPROVED** [review]
LGTM, thanks!

**Review by gsmet — APPROVED** [review]
I squashed the 2 commits.

**Comment by gsmet** [comment]
Superseded by https://github.com/quarkusio/quarkus/pull/9041 .

**Pull request closed without merging**